src/config.ts

```diff
@@ -26,45 +26,30 @@ export const isTokenMetadata = (metadata: any): metadata is TokenMetadata =>
 export const isErc20Metadata = (metadata: any): metadata is ERC20Metadata =>
   metadata.decimals && isTokenMetadata(metadata);
 
-export type SyntheticConfig = TokenMetadata & {
+export type SyntheticConfig = Partial<TokenMetadata> & {
   type: TokenType.synthetic | TokenType.syntheticUri;
 };
 export type CollateralConfig = {
-  type: TokenType.collateral | TokenType.collateralUri;
-  token: string;
-};
-export type NativeConfig = {
-  type: TokenType.native;
+  type: TokenType.collateral | TokenType.collateralUri | TokenType.native;
+  token?: string; // no token implies native collateral
```

Comment:
nit: Maybe `token not required for native collateral`

Just b/c the type is not implied by the presence/absence of token, it's explicit in the type